test/crud-with-options.test.js

```diff
@@ -536,6 +536,54 @@ describe('crud-with-options', function() {
 
   });
 
+  describe('upsertWithWhere', function() {
+
+    beforeEach(seed);
+
+    it('should allow upsertWithWhere(data, cb)', function(done) {
+      User.upsertWithWhere({ name: 'John Smith' }, function() {
+        User.find({ where: { name: 'John Lennon' }}, function(err, data) {
+          should.not.exist(err);
+          data.length.should.equal(1);
+          User.find({ where: { name: 'John Smith' }}, function(err, data) {
+            should.not.exist(err);
+            data.length.should.equal(0);
+            done();
+          });
+        });
+      });
+    });
+
+    it('should allow upsertWithWhere(where, data, cb)', function(done) {
+      User.upsertWithWhere({ where: { name: 'John Lennon' }}, { name: 'John Smith' }, function(err) {
```

Comment:
I thought that the first argument is already the `where` part of the filter, not a full `filter`. I.e. in this test, I would expect to see

```js
User.upsertWithWhere({ name: 'John Lennon' }, { name: 'John Smith' }, ...);
```

It does not give me high confidence in this test suite. I think we need to have at least two existing records before calling `upsertWithWhere` in order to correctly verify the implementation.